tests/clients/test_experiment.py

```diff
@@ -363,3 +427,68 @@ def test_experiment_client_list_runs_page(mocker):
         schema_mock.return_value,
         params=[("start", 20), ("limit", 10)],
     )
+
+
+def test_log_run_data(mocker):
+    mocker.patch.object(ExperimentClient, "_patch_raw")
+    run_data_schema_mock = mocker.patch(
+        "faculty.clients.experiment.ExperimentRunDataSchema"
+    )
+    run_data_dump_mock = run_data_schema_mock.return_value.dump
+
+    client = ExperimentClient(mocker.Mock())
+    client.log_run_data(
+        PROJECT_ID,
+        EXPERIMENT_RUN_ID,
+        metrics=[METRIC],
+        params=[PARAM],
+        tags=[TAG],
+    )
+
+    run_data_schema_mock.assert_called_once_with()
+    run_data_dump_mock.assert_called_once_with(
+        {"metrics": [METRIC], "params": [PARAM], "tags": [TAG]}
+    )
+    ExperimentClient._patch_raw.assert_called_once_with(
+        "/project/{}/run/{}/data".format(PROJECT_ID, EXPERIMENT_RUN_ID),
+        json=run_data_dump_mock.return_value,
+    )
+
+
+def test_log_run_data_param_conflict(mocker):
+    message = "bad params"
+    error_code = "conflicting_params"
+    json_mock = mocker.Mock()
```

Comment:
I wouldn't have bothered commenting on this if there weren't other changes I'd commented on, but `json_mock` would probably better be `response_mock` in this case, since it's really a mock `requests.Response` object.

Reply:
👍